x/leverage/keeper/grpc_query.go

```diff
@@ -90,6 +90,23 @@ func (q Querier) Borrowed(
 	return &types.QueryBorrowedResponse{Borrowed: sdk.NewCoins(token)}, nil
 }
 
+func (q Querier) BorrowApy(
+	goCtx context.Context,
+	req *types.QueryBorrowApyRequest,
+) (*types.QueryBorrowApyResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+	if req.Denom == "" {
+		return nil, status.Error(codes.InvalidArgument, "invalid denom")
+	}
+
```

Comment:
```suggestion

	if !q.Keeper.IsAcceptedToken(ctx, req.Denom) {
		return nil, status.Error(codes.InvalidArgument, "not accepted Token denom")
	}

```
I added this extra safeguard to the other queries in my PR - updating here to match.